### PR TITLE
BAEL 2260-Guide to EnumMap

### PR DESCRIPTION
**Pull request by bungrudi**

### CODE REVIEW

File: core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+        EnumMap<DayOfWeek, String> activityMapCopy = new EnumMap<>(dayMap);
+        assertThat(activityMapCopy.size()).isEqualTo(2);
+        assertThat(activityMapCopy.get(DayOfWeek.MONDAY)).isEqualTo("Soccer");
+        assertThat(activityMapCopy.get(DayOfWeek.TUESDAY)).isEqualTo("Basketball");
+    }
+
+    @Test
+    public void givenEmptyMap_whenConstructedWithMap_ThenException() {
+        HashMap ordinaryMap = new HashMap();
```
Comment (jzheaux) [inline core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java]: @bungrudi Let's parameterize this and code to an interface, e.g. `Map<DayOfWeek, String> ordinaryMap = new HashMap<>()`

File: core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        assertThatCode(() -> new EnumMap(ordinaryMap)).isInstanceOf(IllegalArgumentException.class).hasMessage("Specified map is empty");
+    }
+
+    @Test
+    public void givenMapWithEntries_whenConstructedWithMap_ThenSucceed() {
+        HashMap ordinaryMap = new HashMap();
```
Comment (jzheaux) [inline core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java]: Same here

File: core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java (hunk 1 above)
Comment (jzheaux) [reply core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java]:
```suggestion
        Map<DayOfWeek, String> ordinaryMap = new HashMap<>();
```

File: core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        EnumMap<DayOfWeek, String> activityMapCopy = new EnumMap<>(dayMap);
+        assertThat(activityMapCopy.size()).isEqualTo(2);
+        assertThat(activityMapCopy.get(DayOfWeek.MONDAY)).isEqualTo("Soccer");
+        assertThat(activityMapCopy.get(DayOfWeek.TUESDAY)).isEqualTo("Basketball");
+    }
+
+    @Test
+    public void givenEmptyMap_whenConstructedWithMap_ThenException() {
+        HashMap ordinaryMap = new HashMap();
+        assertThatCode(() -> new EnumMap(ordinaryMap)).isInstanceOf(IllegalArgumentException.class).hasMessage("Specified map is empty");
```
Comment (jzheaux) [inline core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java]: This line scrolls horizontally. Could we please break it up to make it easier for our readers?

```suggestion
        assertThatCode(() -> new EnumMap(ordinaryMap))
          .isInstanceOf(IllegalArgumentException.class)
          .hasMessage("Specified map is empty");
```

File: core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java (hunk 2 above)
Comment (jzheaux) [reply core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java]:
```suggestion
        Map<DayOfWeek, String> ordinaryMap = new HashMap<>();
```

File: core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        EnumMap<DayOfWeek, String> activityMap = new EnumMap(DayOfWeek.class);
+        activityMap.put(DayOfWeek.WEDNESDAY, "Hiking");
+        activityMap.put(DayOfWeek.THURSDAY, null);
+        assertThat(activityMap.get(DayOfWeek.WEDNESDAY)).isEqualTo("Hiking");
+        assertThat(activityMap.get(DayOfWeek.THURSDAY)).isNull();
+    }
+
+    @Test
+    public void givenMapping_whenContains_thenTrue() {
+        EnumMap<DayOfWeek, String> activityMap = new EnumMap(DayOfWeek.class);
```
Comment (jzheaux) [inline core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java]: Personally, I'd prefer

```suggestion
        Map<DayOfWeek, String> activityMap = new EnumMap(DayOfWeek.class);
```

in order to demonstrate that the contracts are the same as `Map`.

**Comment by jzheaux** [comment]
@bungrudi Not sure if you get the other pings when I do a review, so I'll leave a comment here, too. Ping me when you've made the PR changes. Thanks!

**Comment by bungrudi** [comment]
@jzheaux I've made the changes.

**Pull request merged**